src/saltext/vmware/states/esxi.py

```diff
@@ -927,11 +977,20 @@ def advanced_config(
         service_instance=service_instance,
     )
     if __opts__["test"]:
-        ret["result"] = None
-        ret["changes"] = {"new": {}}
-        for host in esxi_config_old:
-            ret["changes"]["new"][host] = f"{name} will be set to {value}"
-        ret["comment"] = "These options are set to change."
+        if config_input:
+            changes = {}
+            for host in esxi_config_old:
+                changes[host] = salt.utils.data.recursive_diff(esxi_config_old[host], config_input["advanced_options"])["new"]
+            ret = {"name": name, "result": True,
+                       "comment": config_input["advanced_options"], "changes": changes}
+            return ret
+        else:
+            ret["result"] = None
+            ret["changes"] = {"new": {}}
+            for host in esxi_config_old:
+                ret["changes"]["new"][host] = f"{name} will be set to {value}"
+                ret["changes"]["old"][host] = f"{name} was {esxi_config_old[host][name]}"
+            ret["comment"] = "These options are set to change."
```

Comment:
**comment**: based on our discussion there are couple of different approaches we can make to extend this - what we *may* want to do is detect which approach the user wants based on what the config values look like and then either manage multiple configs, or just the single one. Of course if just managing a single one isn't that useful, we can go ahead and deprecate that old approach.

Also feel free to update/create new modules if the existing ones either have a bad API or they're too slow because we could get all the data in one call vs several.

Reply:
As suggested in call, we created a new function, and if the new one ends up being more useful we can deprecate old one